userprofile/views.py

```diff
@@ -34,10 +36,30 @@ def _get_user_profiles_from_search(users: [User], search: str):
     """
 
     search_matches = []
+
+    def check_if_user_is_match(user, match_score_min_value):
+        match_score_first_name = fuzz.token_set_ratio(user.get_short_name(), search)
+        match_score_last_name = fuzz.token_set_ratio(user.last_name, search)
+        if (
+            match_score_first_name > match_score_min_value
+            and user not in chain.from_iterable(search_matches)
+        ):
+            search_matches.append((match_score_first_name, user))
+        if (
+            match_score_last_name > match_score_min_value
+            and user not in chain.from_iterable(search_matches)
+        ):
+            search_matches.append((match_score_last_name, user))
```

Comment:
I think this can be refactored a little:
- The `user not in` check can be moved to the top to avoid calculating match score for already added users. This assumes that you don't want to update the score if it is higher than the existing one (which I think is fine for the current setup).
- The match score should be the max of first name score and last name score, right?
```suggestion
        if user in chain.from_iterable(search_matches):
            return
        match_score = max(
            fuzz.token_set_ratio(user.get_short_name(), search),
            fuzz.token_set_ratio(user.last_name, search)
        )
        if match_score > match_score_min_value:
            search_matches.append((match_score, user))
```

Reply:
These edits look good, I agree with the changes you have made.

Reply:
Actually, I don't think taking the highest score of firstname or lastname is the best way to do this. This will mean that searching both names will have no effect surely?